src/cargo/ops/cargo_package/vcs.rs

```diff
@@ -228,6 +262,112 @@ fn git(
     }
 }
 
+/// Helper to collect dirty statuses for a single repo.
+/// `relative_package_root` is `Some` if the root is a sub-directory of the workdir.
+/// Returns the dirty files outside `relative_package_root`.
+#[must_use]
+fn collect_statuses(
+    repo: &gix::Repository,
+    workdir: &Path,
+    relative_package_root: Option<&Path>,
+    dirty_files: &mut Vec<PathBuf>,
+) -> CargoResult<Vec<PathBuf>> {
```

Comment:
nit: May be just personal taste. I feel like it is a bit inconsistent that `dirty_files` is collected via an in-out parameter, while `dirty_files_outside_of_package_root` is a return value. Could we make them both a parameter, and have maybe one or two lines of comment to explain their difference?